src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -517,6 +518,34 @@ std::pair<time_t, time_t> IMergeTreeDataPart::getMinMaxTime() const
     return {};
 }
 
+bool IMergeTreeDataPart::enabledMaxMinTimeOfDataInsertion() const
+{
+    return (*storage.getSettings())[MergeTreeSetting::allow_generate_min_max_data_insert_file];
+}
+
+time_t IMergeTreeDataPart::getMinTimeOfDataInsertion() const
+{
+    if (!(*storage.getSettings())[MergeTreeSetting::allow_generate_min_max_data_insert_file])
+        return 0;
+
+    if (min_time_of_data_insert.has_value())
+    {
+        return *min_time_of_data_insert;
+    }
+    return modification_time;
```

Comment:
We return `modification_time` as a fallback but in `PartsComparatorByOldestData` we rely on `getMinMaxTime()` as a fallback. 
Can you clarify precedence between `min_time_of_data_insert`, `modification_time` and `getMinMaxTime()` ?